backend/libinput/pointer.c

```diff
@@ -84,6 +84,51 @@ void handle_pointer_button(struct libinput_event *event,
 	wlr_signal_emit_safe(&wlr_dev->pointer->events.button, &wlr_event);
 }
 
+static enum wlr_axis_source axis_source_to_wlr(
+		enum libinput_pointer_axis_source src) {
+	switch (src) {
+	case LIBINPUT_POINTER_AXIS_SOURCE_WHEEL:
+		return WLR_AXIS_SOURCE_WHEEL;
+	case LIBINPUT_POINTER_AXIS_SOURCE_FINGER:
+		return WLR_AXIS_SOURCE_FINGER;
+	case LIBINPUT_POINTER_AXIS_SOURCE_CONTINUOUS:
+		return WLR_AXIS_SOURCE_CONTINUOUS;
+	case LIBINPUT_POINTER_AXIS_SOURCE_WHEEL_TILT:
+		return WLR_AXIS_SOURCE_WHEEL_TILT;
+	default:
+		abort();
+	}
+}
+
+static enum wlr_axis_orientation axis_orientation_to_wlr(
+		enum libinput_pointer_axis orientation) {
+	switch (orientation) {
+	case LIBINPUT_POINTER_AXIS_SCROLL_VERTICAL:
+		return WLR_AXIS_ORIENTATION_VERTICAL;
+	case LIBINPUT_POINTER_AXIS_SCROLL_HORIZONTAL:
+		return WLR_AXIS_ORIENTATION_HORIZONTAL;
+	default:
```

Comment:
I usually like to omit the `default` if we handle all the enum items - this allows the compiler to check at compilation if we're missing one case because libinput has added a new one.